### PR TITLE
Reconcile the two SCI_GETSELTEXT queries in NPP source to the changed return value in Scintilla5

### PR DESCRIPTION
**Pull request by shriprem**
Fixes: #11639 - [NotePad++ 8.4 Highlights every Instance of the first word in the document after clicking HELP | ABOUT](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/11639)
Fixes: #11671 - [[Regression] "Copy Binary Content" crashes NPP if there is no selection](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/11671)

The following change is [documented ](https://sourceforge.net/p/scintilla/code/ci/483efdbe6facf2d9a4f10a8ec1f049bfae7723ae/tree/doc/ScintillaHistory.html?diff=2a700fc24c59a47fa6b29b0bb6c556dcc39c5b31)for Scintilla5:
>When calling SCI_GETTEXT, SCI_GETSELTEXT, and SCI_GETCURLINE with a NULL buffer argument to discover the length that should be allocated, do not include the terminating NUL in the returned value. The value returned is 1 less than previous versions of Scintilla. Applications should allocate a buffer 1 more than this to accommodate the NUL. The wParam (length) argument to SCI_GETTEXT and SCI_GETCURLINE also omits the NUL. This is more consistent with other APIs.

This change accounts for the issues reported in #11639 & #11671. This PR makes the identical minor changes needed to fix both these issues.

### CODE REVIEW

**Comment by Yaron10** [comment]
@shriprem,

👍 
Great. Thank you.

File: PowerEditor/src/NppCommands.cpp [hunk 1]
```diff
@@ -373,8 +373,8 @@ void Notepad_plus::command(int id)
 		case IDM_EDIT_COPY_BINARY:
 		case IDM_EDIT_CUT_BINARY:
 		{
-			size_t textLen = _pEditView->execute(SCI_GETSELTEXT, 0, 0) - 1;
-			if (!textLen)
+			size_t textLen = _pEditView->execute(SCI_GETSELTEXT, 0, 0);
+			if (textLen < 1)
 				return;
```
[the remark this answers is not in the document]
Comment (donho) [reply PowerEditor/src/NppCommands.cpp:378]: > if (textLen < 1) is a more general check than just comparing with zero

How come it's a more general check?

[the remark this answers is not in the document]
Comment (shriprem) [reply PowerEditor/src/NppCommands.cpp:378]: Not for this case where `textLen` is of type `size_t`.

But this comment applies in the next case, `textlen` is declared as `auto`. The compiler actually deduces it to be `int` type. To test this, I put the `- 1` back on line 3163 but changed line 3164 to be `if (textLen < 1)`. That alternative change was able to resolve the issue reported in #11639 as well. So, if line 3164 had been coded as `if (textLen < 1)`, it would have prevented the #11639 issue from manifesting because it would be checking for both 0 (pre-Scintilla5) and -1 (Scintilla5). That's why I called it a more general check for that case. Clearly, that claim does not apply for the `size_t textLen' in this case since size_t is unsigned and will wrap around. Pardon my hasty copy-paste.

File: PowerEditor/src/NppCommands.cpp [hunk 2]
```diff
@@ -140,22 +140,22 @@ void Notepad_plus::command(int id)
 			TCHAR currentDir[CURRENTWORD_MAXLENGTH];
 			::SendMessage(_pPublicInterface->getHSelf(), NPPM_GETFULLCURRENTPATH, CURRENTWORD_MAXLENGTH, reinterpret_cast<LPARAM>(currentFile));
 			::SendMessage(_pPublicInterface->getHSelf(), NPPM_GETCURRENTDIRECTORY, CURRENTWORD_MAXLENGTH, reinterpret_cast<LPARAM>(currentDir));
-
```
Comment (donho) [inline PowerEditor/src/NppCommands.cpp:143]: Please restore all unnecessary formatting changes.

**Review by donho — CHANGES_REQUESTED** [review]
Please apply the following changes:
1. restore all unnecessary formatting changes.
2. for the variable of type size_t, restore `if (!textLen)`
3. for the variable of type `auto`, use `if (textLen <= 0)` instead of `if (textLen < 1)`

File: PowerEditor/src/NppCommands.cpp (hunk 2 above)
Comment (shriprem) [reply PowerEditor/src/NppCommands.cpp]: Done. I have just now pushed the commit "Restored white space changes".

**Review by donho — APPROVED** [review]
